torch/_inductor/fx_passes/post_grad.py

```diff
@@ -61,18 +62,24 @@ def post_grad_passes(gm: torch.fx.GraphModule, is_inference: bool):
     if is_inference and config.reordering:
         reorder_for_locality(gm.graph)
 
+    fake_tensor_updater = FakeTensorUpdater(gm.graph)
     if config.pattern_matcher:
         lazy_init()
 
         group_batch_fusion_post_grad_passes(gm.graph)
-        remove_extra_clones(gm.graph)
+        remove_noop_ops(gm.graph)
 
         for patterns in pass_patterns:
             patterns.apply(gm.graph)
         if is_inference:
             inference_patterns.apply(gm.graph)
 
     stable_topological_sort(gm.graph)
+
+    fake_tensor_updater.incremental_update()
+    # Keep this last, since it introduces mutation. Look at
+    # ./fx_passes/README.md for a discussion of mutation invariants.
+    reinplace_scatters(gm.graph)
```

Comment:
Note, pattern matcher prohibits matches across copy_, does not handle index_put_. Nothing needs to be changed today but worth keeping in mind

Reply:
That's why I enforce that this pass has to be last :)